File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int area = 0;
        
        for (int i = 0, j = height.size() - 1; 
             i < j; 
             height[i] < height[j] ? i++ : j--) 
        {
            int newarea = (j - i) * min(height[i], height[j]);
            if (newarea > area) 
            {
                area = newarea;
            }
        }
        
        return area;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    // area = (height[j] - height[i]) * (Math.min(height[j], height[i]));
    var start = 0;
    var end = height.length - 1;
    var max = 0;
    while (start <= end) {
        var area = (end - start) * (Math.min(height[start], height[end]));
        max = Math.max(max, area);
        if (height[start] < height[end]) {
            start++;
        } else if (height[start] > height[end]) {
            end--;
        } else {
            start++;
            end--;
        }
    }
    return max;
};